dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/utils/ProgramBuilder.java

```diff
@@ -347,11 +347,11 @@ public void initVirLocEqLocAliasProxy(String leftName, String rightName){
     public void addSwwPairThreads(int threadId0, int threadId1) {
         Thread thread0 = (Thread) getFunctionOrError(threadId0);
         Thread thread1 = (Thread) getFunctionOrError(threadId1);
-        thread0.optSyncSet.ifPresentOrElse(
+        thread0.getOptSyncSet().ifPresentOrElse(
                 syncSet -> syncSet.add(thread1),
-                () -> thread0.optSyncSet = Optional.of(new HashSet<>(Collections.singletonList(thread1))));
-        thread1.optSyncSet.ifPresentOrElse(
+                () -> thread0.setOptSyncSet(new HashSet<>(Collections.singleton(thread1))));
```

Comment:
I don't think you need the `new HashSet<>` since `Collections.singleton()` should already return a set

Reply:
The `syncSet` is initialized inside the `Thread` constructor in the newest implementation.

Reply:
If the set is not gonna get mutated later, just use `Set.of` to create singletons.
If the set may get mutated later on, use `new HashSet(Set.of(xyz))`.
`Collections.singleton` is unnecessarily verbose, there is no reason to ever use it.